accio-tests/src/test/java/io/accio/testing/bigquery/TestRefreshCache.java

```diff
@@ -41,14 +47,30 @@ protected Optional<String> getAccioMDLPath()
     public void testRefreshFrequently()
             throws InterruptedException
     {
-        // manually reload cache
-        cacheManager.get().createTaskUtilDone(accioMDL.get());
-        // We have one cached table and the most tables existing in duckdb is 2
-        assertThat(queryDuckdb("show tables").size()).isLessThan(3);
-        for (int i = 0; i < 50; i++) {
-            Thread.sleep(200);
-            assertThat(queryDuckdb("show tables").size()).isLessThan(3);
+        AccioMDL mdl = accioMDL.get();
+        CatalogSchemaTableName revenueName = catalogSchemaTableName(mdl.getCatalog(), mdl.getSchema(), "RefreshFrequently");
+        TaskInfo original = cacheManager.get().listTaskInfo(mdl.getCatalog(), mdl.getSchema()).join().stream()
+                .filter(taskInfo -> taskInfo.getCatalogSchemaTableName().equals(revenueName))
+                .findAny().orElseThrow(AssertionError::new);
+        Thread.sleep(6000);
+        // make sure the end time will be changed.
+        TaskInfo refreshed = getTaskInfoUntilDifferent(original);
+        assertThat(original.getEndTime()).isBefore(refreshed.getEndTime());
+    }
+
+    private TaskInfo getTaskInfoUntilDifferent(TaskInfo taskInfo)
```

Comment:
If we are using a active check loop with short sleep, then L55 (sleep 6s) could be removed.
And we should add a timeout for this method to avoid potential infinite while loop.